op-node/p2p/store/serialize.go

```diff
@@ -0,0 +1,27 @@
+package store
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	pool "github.com/libp2p/go-buffer-pool"
+)
+
+func serializeScoresV0(scores PeerScores) ([]byte, error) {
+	var b pool.Buffer
```

Comment:
If we don't return this buffer `b`, will it return to the global buffer pool (the default when left unspecified)? The docs say it frees memory on Read, but the `.Bytes()` function isn't a read; it exposes the buffered data without copy, and the buffer cannot be returned to the pool safely/automatically (since the returned slice might theoretically be mutated). In general I'm against using `sync.Pool` without comments, because it's easy to miss a `.Put` otherwise. The libp2p buffer pool seems to be the same.

Docs: https://pkg.go.dev/github.com/libp2p/go-buffer-pool#Buffer

Reply:
I hope it's safe - I followed the pattern that the metadata store is using but the unpooled version should be fine and is definitely safer so makes sense.